audio_position is a GstClockTime, which is typedef'd as guint64, not gsize.
Use GLib's format macros to do this portably.

--- origsrc/gst-plugins-espeak-0.3.3/src/espeak.c	2009-04-29 14:48:43.000000000 -0500
+++ src/gst-plugins-espeak-0.3.3/src/espeak.c	2010-03-07 03:06:35.530608400 -0600
@@ -346,7 +346,7 @@ play(Econtext *self, Espin *spin, gsize 
     spin->sound_offset += size_to_play;
     spin->events_pos += 1;
 
-    GST_DEBUG("out=%p size_to_play=%zd tell=%zd ts=%zd dur=%zd",
+    GST_DEBUG("out=%p size_to_play=%zd tell=%zd ts=%"G_GUINT64_FORMAT" dur=%"G_GUINT64_FORMAT,
             GST_BUFFER_DATA(out), size_to_play, spin->sound_offset,
             GST_BUFFER_TIMESTAMP(out), GST_BUFFER_DURATION(out));
 
